xbmc/platform/win32/storage/discs/Win32DiscDriveHandler.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "Win32DiscDriveHandler.h"
+
+#include "utils/log.h"
+
+#include "platform/win32/WIN32Util.h"
+
+#include <memory>
+
+std::shared_ptr<IDiscDriveHandler> IDiscDriveHandler::CreateInstance()
+{
+  return std::make_shared<CWin32DiscDriveHandler>();
+}
+
+DriveState CWin32DiscDriveHandler::GetDriveState(const std::string& devicePath)
+{
+  DriveState driveState = DriveState::DRIVE_NOT_READY;
+  auto status = static_cast<CWin32DiscDriveState>(CWIN32Util::GetDriveStatus(devicePath));
+  switch (status)
+  {
+    case CWin32DiscDriveState::DRIVE_ERROR:
+      driveState = DriveState::DRIVE_NOT_READY;
+      break;
+    case CWin32DiscDriveState::NO_MEDIA:
+      driveState = DriveState::DRIVE_CLOSED_NO_MEDIA;
+      break;
+    case CWin32DiscDriveState::TRAY_OPEN:
+      driveState = DriveState::DRIVE_OPEN;
+      break;
+    case CWin32DiscDriveState::MEDIA_ACCESSIBLE:
+      driveState = DriveState::DRIVE_CLOSED_MEDIA_PRESENT;
+      break;
+    default:
+      CLog::LogF(LOGWARNING, "Unknown/unhandled drive state interpreted as DRIVE_NOT_READY");
+      break;
+  }
+  return driveState;
+}
+
```

Comment:
remove

Reply:
comes from my python head, in PEP8 there are a lot of cases where you're advised to keep 2 empty lines. Would be nice if this kind of stuff could be detected by clang-format (if that's even possible)

Reply:
i dont know there might be a good reason to have it set to 2
https://github.com/xbmc/xbmc/blob/e47331c6308b7f7f9760c8813c4f70fc150d4df1/.clang-format#L62

Reply:
Hum added by @Rechi in https://github.com/xbmc/xbmc/commit/9cf98ee7d40554642024c3f9550e3eece209403f
Might be related to objective c. No idea :)

Reply:
I can't remember why I changed it, but I think it was to keep clang-format changes minimal to existing code.
If preferred it chan be changed back to 1, IMO.